pkg/engine/diff.go

```diff
@@ -232,25 +232,32 @@ func PrintObject(
 	}
 }
 
-func massageStackPreviewAdd(p resource.PropertyValue) {
+func massageStackPreviewAdd(p resource.PropertyValue) resource.PropertyValue {
```

Comment:
Nit: It might not be consistent with the other functions in the file, but if you have the context of what this function does right now, would you mind adding a comment on it? TBH I have no idea what `messageStackPreviewAdd` is doing, and what it it is supposed to return. (Does it return the same input? Is it returning a copy of its input? etc.)

Just a suggestion to make it easier for the next person who needs to update this code.